scripts/t212_fetch.py

```
@@ -0,0 +1,326 @@
+
+def make_headers(api_key: str, api_secret: str) -> dict:
+    """Creates Basic Auth headers for the API requests."""
+    creds = base64.b64encode(f"{api_key}:{api_secret}".encode()).decode()
+    return {"Authorization": f"Basic {creds}"}
+
+
+def safe_get(url: str, headers: dict) -> requests.Response:
+    """Wrapper for GET requests that handles 429 Rate Limiting."""
+    while True:
+        print(f"  [GET] {url}")
+        resp = requests.get(url, headers=headers)
```

Ruff (0.15.9)

[error] Probable use of `requests` call without timeout

(S113)